### PR TITLE
cni-plugin: fix plugin execution broken by ambient

### PR DESCRIPTION
**Pull request by PlatformLC**
**Please provide a description of this PR:**
cni-plugin: fix plugin execution broken by ambient

cni cmdADD will not take any effect for legacy sidecar mode while cni configuration enabling ambient.
Fortunately/unfortunately, redirection still works in sidecar because we have the repair/reconcile mechanism for sidecar mode.

### CODE REVIEW

**Comment by bleggett** [comment]
This is a miss on my part from merging the internally-separate containers back into one CNI daemonset - ty @PlatformLC!

File: cni/pkg/plugin/plugin.go [hunk 1]
```diff
@@ -232,8 +232,6 @@ func doAddRun(args *skel.CmdArgs, conf *Config) error {
 		} else {
```
Comment (bleggett) [inline cni/pkg/plugin/plugin.go:232]: I still think we should have the `return` here - just put it after `PushCNIEvent != err` - if we configure the pod for ambient we do not need to perform the subsequent sidecar-only logic or related checks, and can bail early.

**Review by bleggett — CHANGES_REQUESTED** [review]

File: cni/pkg/plugin/plugin.go (hunk 1 above)
Comment (PlatformLC) [reply cni/pkg/plugin/plugin.go:233]: Thanks for reviewing it. Agreed and updated it.

**Review by bleggett — APPROVED** [review]
LGTM and TY!

(note to self, need more tests around `sidecar-and-ambient together` scenarios)

**Pull request merged**